lib/state/ui/reducer.ts

```diff
@@ -22,4 +26,38 @@ const visiblePanes = (state = defaultVisiblePanes, { type, show }) => {
   return state;
 };
 
-export default combineReducers({ filteredNotes, visiblePanes });
+const defaultNote: T.NoteEntity | null = null;
+const setSelectedNote = (state = defaultNote, action) => {
+  switch (action.type) {
+    case 'App.notesLoaded': {
+      if (!state) {
+        return action.notes[0] || null;
+      }
+
+      const note = action.notes.find(({ id }) => id === state.id);
+      return note || state;
```

Comment:
the code as written is fine but now that I'm looking at it there's also the collapsed version

```ts
return action.notes.find(({ id }) => id === state.id) || null;
```

though if what you wrote earlier is the case and we need to check for `notes.length` obviously we'll need more code.

if we're safe (which we can verify through the type system)…

```ts
case 'App.notesLoaded':
  return state
    ? (action.notes.find(({id}) => id === state.id) || state)
    : (action.notes[0] || null);
```